### PR TITLE
x86: Add apu6 support

### PR DESCRIPTION
**Pull request by pprindeville**
The [APU6](https://www.pcengines.ch/apu6b4.htm) is an SBC from PC Engines based on the AMD GX-412TC SOC, and is a quite capable 4-port router with 2GB of DRAM, a mSATA slot, mPCI-e slots, battery backed up CMOS RTC, etc.  The first ethernet port is SFP which is handy for fiber WAN termination.

We also add APU4 support, which is already in the kernel, but there is no board description.  The APU4 resembles the APU6, but has no SFP cage.

### CODE REVIEW

**Comment by pprindeville** [comment]
cc @ynezz @chunkeey @hauke @PolynomialDivision @feckert @corna @Hurricos @aparcar @rsalvaterra

**Comment by aparcar** [comment]
Did you send the kernel stuff upstream? The other commit looks good to me, could you mention on the mailing list that your previous patched are superseded?

**Comment by pprindeville** [comment]
> Did you send the kernel stuff upstream? The other commit looks good to me, could you mention on the mailing list that your previous patched are superseded?

I did.  It's... in the pipeline:

https://www.spinics.net/lists/platform-driver-x86/msg36193.html
https://www.spinics.net/lists/linux-x86_64/msg00624.html

**Comment by aparcar** [comment]
Do you mind adding those links to the commit adding the patches? This way we can track in the future how things went.

Apart from that I don't have the device to test it but I'm confident it works fine.

**Comment by pprindeville** [comment]
I'll add the links to patchwork instead.

BTW, Ed Wildgoose asked me to merge the APU5 and APU6 changes into a single PR, so I did.

File: target/linux/x86/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -11,7 +11,7 @@ cisco-mx100-hw)
 	ucidef_set_led_usbport "usb" "USB" "mx100:green:usb" "1-1-port2"
 	ucidef_set_led_default "diag" "DIAG" "mx100:green:tricolor" "1"
 	;;
-pc-engines-apu1|pc-engines-apu2|pc-engines-apu3)
+pc-engines-apu1|pc-engines-apu2|pc-engines-apu3|pc-engines-apu4|pc-engines-apu6)
```
Comment (aparcar) [inline target/linux/x86/base-files/etc/board.d/01_leds:14]: Shouldn't we add v5 here too?

Comment (pprindeville) [reply target/linux/x86/base-files/etc/board.d/01_leds]: Yes, good point.  Just need to confirm that it has 3 Ethernet ports.

File: target/linux/x86/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -23,9 +23,12 @@ cisco-mx100-hw)
 	ucidef_set_network_device_path "eth11" "pci0000:00/0000:00:01.1/0000:02:00.2"
 	ucidef_set_interfaces_lan_wan "mgmt eth2 eth3 eth4 eth5 eth6 eth7 eth8 eth9 eth10 eth11" "wan"
 	;;
-pc-engines-apu1|pc-engines-apu2|pc-engines-apu3)
+pc-engines-apu1|pc-engines-apu2|pc-engines-apu3|pc-engines-apu5)
```
Comment (obsy) [inline target/linux/x86/base-files/etc/board.d/02_network:26]: APU5? Any link?

Comment (pprindeville) [reply target/linux/x86/base-files/etc/board.d/02_network:26]: Alas the APUv5 is a bespoke design for a channel partner, so there’s no public page for it.

File: target/linux/x86/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (pprindeville) [reply target/linux/x86/base-files/etc/board.d/01_leds]: Actually, the APU5 only has two ethernet ports, so adjusting...

**Comment by pprindeville** [comment]
Any other reviewers?  @nbd168 @jow- @dangowrt @hauke @kaloz @NeoRaider

**Comment by aparcar** [comment]
Thanks @ynezz

**Comment by ynezz** [comment]
https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/x86/patches-5.15/103-pcengines_apu6_platform.patch;h=0ef3c0c087c0d308dfd506b99dd5ac110a440e86;hb=5ffa8d06c87fea2ce2159d2933c178b2ffe8399a ?

**Pull request closed without merging**